index.js

```diff
@@ -38,6 +38,10 @@ var validate = module.exports = function (name) {
     errors.push('name cannot start with an underscore')
   }
```

Comment:
With your new conditional check we can get rid of the checks between line# 33 and 39

Reply:
Actually no. Because for the old packages, we should throw the error only when the package name is starting with `period` and `underscore`. If we remove those changes, it will start throwing the error for all the special characters.